### PR TITLE
[wayland] Using `build_requires` (Conan 1.52)

### PR DESCRIPTION
**Pull request by franramirez688**
Specify library name and version: **wayland/***
**Requires**: Conan 1.52 version

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wayland/all/conanfile.py [hunk 1]
```diff
@@ -79,22 +81,23 @@ def generate(self):
...
+        pc = PkgConfigDeps(self)
         if Version(self.version) >= "1.18.91":
             tc.project_options["scanner"] = True
+            if cross_building(self):
+                # Only wayland is the build_require. There is no need to apply suffixes.
+                pc.build_context_activated = ["wayland"]
+                tc.project_options["build.pkg_config_path"] = self.generators_folder
+            elif self.dependencies["expat"].is_build_context:  # wayland is being built as build_require
+                # If wayland is the build_require, all its dependencies are treated as build_requires
+                pc.build_context_activated = ["libffi", "expat", "libxml2", "libiconv", "zlib"]
```
Comment (jgsogo) [inline recipes/wayland/all/conanfile.py:93]: it cannot be hardcoded, the list of transitive dependencies will be different depending on the resolved graph.

**Comment by jwillikers** [comment]
@franramirez688 Conan 1.52.0 is now available in CCI if you want to open this now. I've just opened PR #13363 which incorporates your changes with a few other small fixes, too.

**Comment by franramirez688** [comment]
> @franramirez688 Conan 1.52.0 is now available in CCI if you want to open this now. I've just opened PR #13363 which incorporates your changes with a few other small fixes, too.

@jwillikers So sorry for the delay, I had some (good) issues these weeks before but I'm back again. I just saw that the changes are already there, right? 👏  Then, I guess that we can close this PR. Thanks!

**Comment by jwillikers** [comment]
@franramirez688 Yes, it can be closed. There is still some stuff that needs worked out with how dependencies of packages which are put into the build context are missing. See conan-io/conan#12342 and #13668 for details.

**Pull request closed without merging**